tower-util/src/ready.rs

```diff
@@ -8,27 +8,114 @@ use std::{
 };
 use tower_service::Service;
 
-/// Future yielding a `Service` once the service is ready to process a request
+/// A future that yields the service when it is ready to accept a request.
 ///
-/// `Ready` values are produced by `ServiceExt::ready`.
-pub struct Ready<'a, T, Request> {
-    inner: &'a mut T,
+/// `ReadyOneshot` values are produced by `ServiceExt::ready_oneshot`.
+pub struct ReadyOneshot<T, Request> {
```

Comment:
Do we want a manual clone impl if T is clone?

Reply:
My first instinct says no — it should be rare that you'd want to clone a `ReadyOneshot` as opposed to cloning the underlying `Service`. If you did, that's _probably_ not what you meant to do.